optimum/intel/generation/modeling.py

```diff
@@ -186,8 +277,15 @@ def _from_transformers(
             "force_download": force_download,
         }
 
-        model = TasksManager.get_model_from_task(task, model_id, **model_kwargs)
-        model.config.return_dict = False
+        if isinstance(model_id, str):
+            model = TasksManager.get_model_from_task(task, model_id, **model_kwargs)
+            model.config.return_dict = False
+        elif isinstance(model_id, torch.nn.Module):

```

Comment:
I don't think we should leave the possibility to give all the options to give as input the model path, an `nn.Module` and a `RecursiveScriptModule` but to only give the model path. This could be confusing to users in my opinion, also this doesn't respect the format we have for Optimum `OVModel` and `ORTModel`